src/webportal/src/app/job-submission/components/tools/ssh-generator.jsx

```diff
@@ -0,0 +1,193 @@
+    setSshKeys(keys);
+  };
+
+  const handleCancel = () => {
+    hide();
+  };
+
+  const handleConfirm = () => {
+    onSshKeysChange(sshKeys);
+    hide();
+  };
+
+  const downloadAsFile = (content, saveName, ev) => {
+    const blob = new Blob([content], {
+      type: 'application/octet-stream,charset=UTF-8',
+    });
+    openDownloadDialog(blob, saveName);
+  };
+
+  const openDownloadDialog = (url, saveName) => {
+    if (typeof url === 'object' && url instanceof Blob) {
+      url = URL.createObjectURL(url);
+    }
+    const aLink = document.createElement('a');
+    aLink.href = url;
+    aLink.download = saveName || '';
+    aLink.click();
+  };
+
+  return (
+    <Dialog
+      hidden={!isOpen}
+      onDismiss={handleCancel}
+      dialogContentProps={{
+        type: DialogType.normal,
+        title: (
+          <span
+            className={c(t.mb2, t.fw6, FontClassNames.semibold)}
+            style={{ fontSize: FontSizes.icon }}
+          >
+            SSH Key Generator
+          </span>
+        ),
+      }}
+      minWidth={900}
+      maxWidth={900}
+      modalProps={{
+        isBlocking: false,
+        styles: { main: { maxWidth: 900 } },
+      }}
+    >
+      <Hint>
+        Please download SSH private key then click <b>Use Public Key</b> button
+        to use this key pair in job. Then after job submitted, you can ssh to
+        job containers as user root with the downloaded private key through
+        container ip and ssh port. Try using{' '}
+        <code>
+          {'ssh -i <private key path> -p <ssh port> root@<container ip>'}
+        </code>{' '}
+        to ssh to job container.
+      </Hint>
+
+      <Separator>SSH key pair</Separator>
+
+      <Stack
+        horizontal
+        tokens={{ childrenGap: 50 }}
+        styles={{ root: { width: 860 } }}
+      >
+        <Stack tokens={{ childrenGap: 10 }} grow={1}>
+          <TextField
+            label='Private Key'
+            multiline
+            rows={20}
+            readonly
+            defaultValue={isNil(sshKeys) ? '' : sshKeys.private}
+          />
+          <DefaultButton
+            onClick={ev => {
+              setDownloadedPriKey(true);
+              downloadAsFile(sshKeys.private, 'id_rsa_pai', ev);
+            }}
+          >
+            Download Private Key
+          </DefaultButton>
+        </Stack>
+        <Stack tokens={{ childrenGap: 10 }} grow={1}>
+          <TextField
+            label='Public Key'
+            multiline
+            rows={20}
+            readonly
+            defaultValue={isNil(sshKeys) ? '' : sshKeys.public}
+          />
+          <DefaultButton
+            onClick={ev => downloadAsFile(sshKeys.public, 'id_rsa_pai.pub', ev)}
+          >
+            Download Public Key
+          </DefaultButton>
+        </Stack>
+      </Stack>
+
+      <Separator></Separator>
+
+      <Stack
+        horizontal
+        horizontalAlign='space-evenly'
+        tokens={{ childrenGap: 150 }}
```

Comment:
Please use Stack's `gap` prop and predefined spacing values (s1/m/l1/l2) instead of `tokens` prop and custom values

Reply:
Done